init.lua

```diff
@@ -160,22 +177,22 @@ minetest.register_chatcommand("geoip", {
 
 		local ip = minetest.get_player_ip(param)
 
-		if not ip then
-			return true, "no ip available!"
-		end
-
-		geoip.lookup(ip, function(result)
-			local txt = format_result(result)
-
-			if not txt then
-				minetest.chat_send_player(name, "Geoip error: " .. (result.description or "unknown error"))
-				return
+		if ip then
+			-- go through lookup if ip is available, this might still return cached result
+			geoip.lookup(ip, function(result)
+				report_result(name, param, result)
+			end, param)
+		else
+			for _, result in pairs(cache) do
+				for playername in pairs(result.players) do
+					if playername == param then
+						report_result(name, param, result)
+						return
+					end
+				end
```

Comment:
```suggestion
				if result.players[param] then
					report_result(name, param, result)
					return
				end
```
you drunk? this loop is just stupid...